### PR TITLE
Clippy: Enable many style lints

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This pull request enables many of the clippy style lints that to me seem noncontroversial.

You can see the change made by each one in each commit. If one seems like it needs more discussion I'm happy to move it to a new PR.

I also fixed a mistake from an earlier clippy PR where we both denied and allowed a flag (before I decided to enumerate all allows).

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by lschuermann — APPROVED** [review]
These all broadly make sense to me.

File: capsules/extra/src/hmac.rs [hunk 1]
```diff
@@ -147,9 +147,7 @@ impl<
                             })
                         })
                         .unwrap_or(Err(ErrorCode::RESERVE));
-                    if ret.is_err() {
-                        return ret;
-                    }
+                    ret?;
```
Comment (ppannuto) [inline capsules/extra/src/hmac.rs:150]: I know this is super-Rusty, but I need my IDE to quintuple the font size of the `?` or something — hiding an early-exit from a function in a single character is not my favorite design decision.

File: capsules/extra/src/net/network_capabilities.rs [hunk 2]
```diff
@@ -49,7 +49,7 @@ impl AddrRange {
                 let full_bytes: usize = prefix_len / 8;
                 let remainder_bits: usize = prefix_len % 8;
                 // initial bytes -- TODO: edge case
-                if &allowed_addr.0[0..full_bytes] != &addr.0[0..full_bytes] {
+                if allowed_addr.0[0..full_bytes] != addr.0[0..full_bytes] {
```
Comment (ppannuto) [inline capsules/extra/src/net/network_capabilities.rs:52]: The fact that these two expressions are equivalent... this is why borrows are so hard to teach :/

File: chips/lowrisc/src/spi_host.rs [hunk 3]
```diff
@@ -361,15 +361,14 @@ impl<'a> SpiHost<'a> {
...
 
-        if txfifo_num_bytes > u8::MAX as u32 {
-            num_transfer_bytes = u8::MAX as u32;
+        let num_transfer_bytes: u32 = if txfifo_num_bytes > u8::MAX as u32 {
+            u8::MAX as u32
         } else {
-            num_transfer_bytes = txfifo_num_bytes;
-        }
+            txfifo_num_bytes
+        };
```
Comment (ppannuto) [inline chips/lowrisc/src/spi_host.rs:371]: The comment for this variable needs to move too

File: kernel/src/process_standard.rs [hunk 4]
```diff
@@ -578,16 +578,12 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
...
-                return None;
-            }
+            )?;
 
             for region in self.mpu_regions.iter() {
                 if region.get().is_none() {
-                    region.set(new_region);
-                    return new_region;
+                    region.set(Some(new_region));
+                    return Some(new_region);
```
Comment (ppannuto) [inline kernel/src/process_standard.rs:586]: This is weird — why did this compile before?

File: kernel/src/utilities/math.rs [hunk 5]
```diff
@@ -115,7 +115,7 @@ fn ln_1to2_series_approximation(x: f32) -> f32 {
     //https://en.wikipedia.org/wiki/Remez_algorithm
     let ln_1to2_polynomial: f32 = -1.741_793_9_f32
         + (2.821_202_6_f32
-            + (-1.469_956_8_f32 + (0.447_179_55_f32 - 0.056_570_851_f32 * x_working) * x_working)
+            + (-1.469_956_8_f32 + (0.447_179_55_f32 - 0.056_570_85_f32 * x_working) * x_working)
```
Comment (ppannuto) [inline kernel/src/utilities/math.rs:118]: Did this just trim a bit off the constant here?

**Review by ppannuto — CHANGES_REQUESTED** [review]
Two screaming into the void comments, three real ones

File: kernel/src/utilities/math.rs (hunk 5 above)
Comment (brghena) [reply kernel/src/utilities/math.rs:118]: That final `1` is beyond the precision of a 32-bit float. Your options are 0.05657085**0**3127 or 0.05657085**4**038. It looks like there are a couple clippy lints that look at floating point literals to ensure that they make sense.

File: capsules/extra/src/hmac.rs (hunk 1 above)
Comment (bradjc) [reply capsules/extra/src/hmac.rs:150]: Yeah, that's the pain point with not having these lints on from the start. The `?` should be on the original function call. But then that changes the types and this is no longer a mechanical change, it's more invasive.

If the developer had gotten this clippy warning, they would have realized they could have used the `?` earlier and we wouldn't have this odd line. This is fixable, but not in the spirit of these clippy prs.

File: kernel/src/process_standard.rs (hunk 4 above)
Comment (bradjc) [reply kernel/src/process_standard.rs:586]: Yeah ok I guess I did the more invasive change here. I'll move this change to its own PR.

File: kernel/src/utilities/math.rs (hunk 5 above)
Comment (ppannuto) [reply kernel/src/utilities/math.rs:118]: On the one hand, that makes sense — on the other hand, curious what it was doing before... always wary about changing magic constants, even in the insignificant bits.

@alexandruradovici , looks like you added this in 8daa35 — any issues here?

Comment (bradjc) [reply kernel/src/utilities/math.rs:118]: As I understand it this is the same as before, Rust was truncating the one.

https://rust-lang.github.io/rust-clippy/master/index.html#/excessive_precision

File: chips/lowrisc/src/spi_host.rs (hunk 3 above)
Comment (bradjc) [reply chips/lowrisc/src/spi_host.rs:371]: Moved.

**Review by ppannuto — DISMISSED** [review]

**Review by lschuermann — DISMISSED** [review]

**Review by hudson-ayers — DISMISSED** [review]

**Review by alevy — DISMISSED** [review]

**Review by ppannuto — APPROVED** [review]

**Pull request merged**